group/assignment6/assignment6.py

```diff
@@ -56,81 +56,4 @@ def compute_efficient_moves(start_state, end_state):
-        
-        if misplaced_car_lot < len(parking):
-            yield parking.move_to_empty_lot(parking.get_car(misplaced_car_lot))
-
-
-def compute_moves_with_constraints(start_state, end_state, constraints):
-    """
-    Computes moves sequence under given constraints.
-    :param start_state: order of cars in the start of the rearrangement
-    :param end_state: order of cars after rearrangement
-    :param constraints: map from the parking lot to a tuple of the allowed cars
-    :yields: move steps. Each move is represented as a tuple with two indeces,
-             the 1st index is the number of lot from which we move the car,
-             the 2nd index is the number of lot to which we move the car
-    """
-    path_finder = PathFinder(tuple(start_state), tuple(end_state), constraints)
-    paths = path_finder.find_all_paths()
-    path = path_finder.decode_path(next(paths))
-    for i in range(1, len(path)):
-        yield compute_move(path[i - 1], path[i])
-
-
-def compute_all_moves(start_state, end_state):
-    """
-    Computes alll possible ways to rearrange the cars from start_state to end_state.
-    :param start_state: order of cars in the start of the rearrangement
-    :param end_state: order of cars after rearrangement
-    :yields: a list of move steps. Each move is represented as a tuple with two indeces,
-             the 1st index is the number of lot from which we move the car,
-             the 2nd index is the number of lot to which we move the car
-    """
-    path_finder = PathFinder(tuple(start_state), tuple(end_state), constraints={}) 
-    paths = path_finder.find_all_paths()
-    for path in paths:
-        decoded_path = path_finder.decode_path(path)
-        moves_sequence = list()
-        for i in range(1, len(decoded_path)):
-            moves_sequence.append(compute_move(decoded_path[i - 1], decoded_path[i]))
-        yield moves_sequence
-    
-
-if __name__ == '__main__':
-    start_state = [1, 2, 0, 3] 
-    end_state = [3, 1, 2, 0]
-
-    print("Computing a moves sequence for the main problem...")
-    moves = list(compute_moves(start_state, end_state))
-    print(moves)
-    
-    print("Computing the shortest moves sequence...")
-    moves = list(compute_efficient_moves(start_state, end_state))
-    print(moves)
-
-    print("Computing a moves sequence under given constraints...")
-    constraints = {0: (1, 3), 1: (0, 1, 2, 3), 2: (0, 1, 2, 3), 3: (0, 1, 2, 3)}
-    moves = list(compute_moves_with_constraints(start_state, end_state, constraints))
-    print(moves)
-
-    print("Computing all moves sequences between two parking states...")
-    moves_sequences = list(compute_all_moves(start_state, end_state))
-    n_sequences = len(moves_sequences)
-    n = min(5, n_sequences)
-    print("Found {} possible moves sequences, printing first {} moves sequences:".format(n_sequences, n)) 
-    for moves in moves_sequences[:n]:
-        print(moves)
+    input_check(start_state, end_state)
```

Comment:
you missed the main fuction. please return it

Reply:
returned